### ISSUE TITLE
Bad state: failed to create store: Cannot open store: another store

### ISSUE THREAD

**Issue by centy** [objectbox/objectbox-dart#703]
### Build info

- objectbox version:"4.0.3"
- Flutter/Dart version: Flutter: 3.27.0 Dart SDK version: 3.6.0 
- Build OS: [Windows 10 ]

### Steps to reproduce
Getting such callstack on Crashlitics.

Fatal Exception: io.flutter.plugins.firebase.crashlytics.FlutterError: Bad state: failed to create store: Cannot open store: another store is still open using the same path: "/data/data/com.gst.color.numbers.pixyfy/app_flutter/objectbox" (OBX_ERROR code 10001)
       at ObjectBoxNativeError.throwMapped(helpers.dart:74)
       at .throwLatestNativeError(helpers.dart:54)
       at .checkObxPtr(helpers.dart:31)
       at Store._checkStorePointer(store.dart:451)
       at .new Store(store.dart:274)
       at .openStore(objectbox.g.dart:271)
       at PictureInfoRepository.performOpen(picture_info_repository.dart:164)

We are initializing DB in main functon.

Scerario look more less like this:

1. User start app and it works
2. App show ad
3. Ad became closed and app is back again --> database is locked

I can not reproduce that, some additional conditions must occurs. This what we see in crashlitics

### Expected behavior
Do not crash or some workaround

### Actual behavior
App not starting

**Comment by greenrobot-team** [comment]
Thanks for reporting!

TL;DR: see #436 for a possible workaround.

A little background: the ObjectBox database runs on the process level. Only some state lives in the Dart VM, but otherwise calls are made to the native implementation.

Somehow it appears that the Dart state can be cleared (e.g. the Dart app terminates) without the native process ending, leaving the ObjectBox database open.

I wonder if using something like an [AppLifecycleListener](https://api.flutter.dev/flutter/widgets/AppLifecycleListener-class.html) and then opening and closing when leaving or entering the detached state would solve this?

**Comment by centy** [comment]
I just wondering if it is possible to detect this case and attach to running instance?

**Comment by greenrobot-team** [comment]
Closing this issue due to inactivity. :zzz: Feel free to comment with more details or submit a new issue.
